src/Startup.cs

```diff
@@ -48,7 +49,13 @@ public void ConfigureServices(IServiceCollection services)
             var sharedAssembly = typeof(CourseDetailsViewComponent).GetTypeInfo().Assembly;
             services.AddMvc(options =>
                 options.Filters.Add(typeof(SearchAndCompareApiExceptionFilter))
-            ).AddApplicationPart(sharedAssembly);
+            ).AddCookieTempDataProvider(options => {
+                options.Cookie.SecurePolicy= CookieSecurePolicy.Always;
+            }).AddApplicationPart(sharedAssembly);
+
+            services.AddAntiforgery(options => {
```

Comment:
We don't want this for dev so it should go into the Configure() method with something along these lines: Cookie.SecurePolicy = hostingEnvironment.IsDevelopment()
            ? CookieSecurePolicy.SameAsRequest
            : CookieSecurePolicy.Always

Reply:
Amended just for you, dev should really be https